metricflow/test/model/test_resolved_where_filter.py

```diff
@@ -0,0 +1,91 @@
+import logging
+
+import pytest
+
+from dbt_semantic_interfaces.objects.filters.where_filter import WhereFilter
+from dbt_semantic_interfaces.objects.time_granularity import TimeGranularity
+from dbt_semantic_interfaces.references import EntityReference
+from metricflow.model.semantic_model import SemanticModel
+from metricflow.plan_conversion.column_resolver import DefaultColumnAssociationResolver
+from metricflow.specs import (
+    ColumnAssociationResolver,
+    LinkableSpecSet,
+    DimensionSpec,
+    EntitySpec,
+    TimeDimensionSpec,
+    ResolvedWhereFilter,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture
+def column_association_resolver(simple_semantic_model: SemanticModel) -> ColumnAssociationResolver:  # noqa: D
+    return DefaultColumnAssociationResolver(simple_semantic_model)
+
+
+def test_dimension_in_filter(  # noqa: D
+    simple_semantic_model: SemanticModel,
+    column_association_resolver: ColumnAssociationResolver,
+) -> None:
+    where_filter = WhereFilter(where_sql_template="{{ dimension('country_latest', entity_path=['listing']) }} = 'US'")
+
+    resolved_where_filter = ResolvedWhereFilter.create_from_where_filter(
+        where_filter=where_filter,
+        column_association_resolver=column_association_resolver,
+    )
+
+    assert resolved_where_filter.where_sql == "listing__country_latest = 'US'"
+    assert resolved_where_filter.linkable_spec_set == LinkableSpecSet(
+        dimension_specs=(
+            DimensionSpec(element_name="country_latest", entity_links=(EntityReference(element_name="listing"),)),
+        ),
+        time_dimension_specs=(),
+        entity_specs=(),
+    )
+
+
+def test_time_dimension_in_filter(  # noqa: D
+    simple_semantic_model: SemanticModel,
+    column_association_resolver: ColumnAssociationResolver,
+) -> None:
+    where_filter = WhereFilter(
+        where_sql_template="{{ time_dimension('created_at', 'month', entity_path=['listing']) }} = '2020-01-01'"
+    )
+
+    resolved_where_filter = ResolvedWhereFilter.create_from_where_filter(
+        where_filter=where_filter,
+        column_association_resolver=column_association_resolver,
+    )
+
+    assert resolved_where_filter.where_sql == "listing__created_at__month = '2020-01-01'"
```

Comment:
Do we want to dunder the month on the end here? Is this because we'll do the dialect-specific DATE_TRUNC() internally and alias the output to the `__month`? I'm worried about that causing issues with optimizer passes, although I think right now it won't be a problem.

I suspect this is a question for later - right now the system works on that rendered dunder so this should match current behavior, right?